tools/pylib/boutdata/data.py

```diff
@@ -266,17 +511,41 @@ class BoutOptionsFile(BoutOptions):
 
     """
 
-    def __init__(self, filename="BOUT.inp", name="root", gridfilename=None, nx=None, ny=None, nz=None):
+    # Get not just the comment, but also the preceeding whitespace
+    COMMENT_REGEX = re.compile(r"(\s*)(#.*)")
+
+    def __init__(
+        self,
+        filename="BOUT.inp",
+        name="root",
+        gridfilename=None,
+        nx=None,
+        ny=None,
+        nz=None,
+    ):
         BoutOptions.__init__(self, name)
+        self.filename = filename
         # Open the file
         with open(filename, "r") as f:
             # Go through each line in the file
             section = self  # Start with root section
+            comments = []
             for linenr, line in enumerate(f.readlines()):
                 # First remove comments, either # or ;
-                startpos = line.find("#")
-                if startpos != -1:
-                    line = line[:startpos]
+                if line.lstrip().startswith("#"):
+                    comments.append(line.strip())
+                    continue
+                if line.strip() == "":
+                    comments.append(line.strip())
+                    continue
+
+                comment_match = self.COMMENT_REGEX.search(line)
+                if comment_match is not None:
+                    line = line.split("#")[0].strip()
+                    comment_whitespace = comment_match.group(1)
+                    inline_comment = comment_match.group(2).strip()
+                else:
+                    inline_comment = None
                 startpos = line.find(";")
```

Comment:
I never use `;` for comments in `BOUT.inp`, but I think we support `;` as equivalent to `#`. Is it possible to treat them both the same here?

Reply:
I somehow just ignored that bit. It looks like there are no input files in the repo that actually use ';' as a comment.

Still, fixed!

Reply:
Probably use of `;` as comments can be deprecated / removed in 5? It was a simpler time of IDL where `;` was first used, but now is not.

Reply:
Can also handle that in this script! Just always write out '#' as the initial comment character